clients/android/app/src/main/java/app/lockbook/loggedin/mainscreen/MainScreenFragment.kt

```diff
@@ -0,0 +1,138 @@
+package app.lockbook.loggedin.mainscreen
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import app.lockbook.R
+import app.lockbook.databinding.FragmentMainScreenBinding
+import app.lockbook.loggedin.listfiles.FilesAdapter
+import app.lockbook.loggedin.newfile.NewFileActivity
+import app.lockbook.loggedin.popupinfo.PopUpInfoActivity
+import app.lockbook.loggedin.texteditor.TextEditorActivity
+import app.lockbook.utils.FileMetadata
+import app.lockbook.utils.RequestResultCodes.NEW_FILE_REQUEST_CODE
+import app.lockbook.utils.RequestResultCodes.POP_UP_INFO_REQUEST_CODE
+import app.lockbook.utils.RequestResultCodes.TEXT_EDITOR_REQUEST_CODE
+
+class MainScreenFragment : Fragment() {
+
+    lateinit var mainScreenViewModel: MainScreenViewModel
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val binding: FragmentMainScreenBinding = DataBindingUtil.inflate(
+            inflater, R.layout.fragment_main_screen, container, false
+        )
+        val application = requireNotNull(this.activity).application
+        val mainScreenViewModelFactory =
+            MainScreenViewModelFactory(application.filesDir.absolutePath)
+        mainScreenViewModel =
+            ViewModelProvider(this, mainScreenViewModelFactory).get(MainScreenViewModel::class.java)
+        val adapter = FilesAdapter(mainScreenViewModel)
+
+        binding.mainScreenViewModel = mainScreenViewModel
+        binding.filesList.adapter = adapter
+        binding.filesList.layoutManager = LinearLayoutManager(context)
+        binding.lifecycleOwner = this
+
+        binding.filesListRefresh.setOnRefreshListener {
+            mainScreenViewModel.refreshFiles()
+            mainScreenViewModel.sync()
```

Comment:
1. Is this in the wrong order? Shouldn't we `refreshFiles` after `sync`?
2. This seems to happen on the `ui` thread, doesn't it?